src/mame/apple/apple2video.cpp

```diff
@@ -865,36 +817,39 @@ void a2_video_device::dhgr_update(screen_device &screen, bitmap_ind16 &bitmap, c
 					}
 					break;
 
-				// RGB 160-wide mode (which has a much simpler VRAM layout)
 				case 4:
-					if (col == 0)
+					// Video-7 RGB with m_rgbmode == 1 (mixed) or 3 (color). In color mode, the card
+					// seems to produce just 7 wide pixels per 4 bytes:
+					//   column & 3 =  0        1        2        3
+					//              nBBBAAAA nDDCCCCB nFEEEEDD nGGGGFFF
+					//
+					// In mixed mode, the Video-7 User's Manual says:
+					//
+					// "When [the MSB is 0] the hardware displays the remaining seven bits as bit-mapped
+					// video; a logic 'one' will instruct the hardware to display the next seven bits
+					// as color pixels. [...] color aberrations will occur at the leading and trailing
+					// edges of the transitions from one mode to the other. The aberrations may be
+					// eliminated by blanking enough bytes at the beginning and end of each transition
+					// to guarantee the four bit integrity of the corresponding color pixels."
+					//
+					// It's impossible to know the nature of the aberrations without hardware to test,
+					// but hopefully this warning means software was designed so that it doesn't matter.
+
+					if (m_rgbmode == 1 && !(vram_row[col+1] & 0x80))  // monochrome
 					{
-						// Center the 480-wide image in the 560-wide display.
-						// Aspect ratio won't be perfect, but it's in range.
-						for (int b = 0; b < 40; b++)
-						{
-							*(p++) = BLACK;
-						}
+						for (int b = 0; b < 7; b++)
```

Comment:
Another style thing here and immediately below: I always use brackets on if statements and for loops, even though they're optional for one-liners.